[SPARK-14301][Examples] Java examples code merge and clean up.

yongtang opened the pull request:
## What changes were proposed in this pull request?

This fix tries to remove duplicate Java code in examples/mllib and examples/ml. The following changes have been made:

```
deleted: ml/JavaCrossValidatorExample.java (duplicate of JavaModelSelectionViaCrossValidationExample.java)
deleted: ml/JavaTrainValidationSplitExample.java (duplicated of JavaModelSelectionViaTrainValidationSplitExample.java)
deleted: mllib/JavaFPGrowthExample.java (duplicate of JavaSimpleFPGrowth.java)
deleted: mllib/JavaLDAExample.java (duplicate of JavaLatentDirichletAllocationExample.java)
deleted: mllib/JavaKMeans.java (merged with JavaKMeansExample.java)
deleted: mllib/JavaLR.java (duplicate of JavaLinearRegressionWithSGDExample.java)
updated: mllib/JavaKMeansExample.java (merged with mllib/JavaKMeans.java)
```
## How was this patch tested?

Existing tests passed.

yongtang committed:
[SPARK-14301][Examples] Java examples code merge and clean up.

This fix tries to remove duplicate Java code in examples/mllib and
examples/ml. The following changes have been made:

deleted: ml/JavaCrossValidatorExample.java (->JavaModelSelectionViaCrossValidationExample.java)
deleted: ml/JavaTrainValidationSplitExample.java (-> JavaModelSelectionViaTrainValidationSplitExample.java)
deleted: ml/JavaSimpleTextClassificationPipeline.java (-> JavaModelSelectionViaCrossValidationExample.java)
deleted: ml/JavaDeveloperApiExample.java (conform to changes in scala/DeveloperApiExample.scala)
deleted: mllib/JavaFPGrowthExample.java (-> JavaSimpleFPGrowth.java)
deleted: mllib/JavaLDAExample.java (-> JavaLatentDirichletAllocationExample.java)
deleted: mllib/JavaKMeans.java (merged with JavaKMeansExample.java)
deleted: mllib/JavaLR.java (-> JavaLinearRegressionWithSGDExample.java)
updated: mllib/JavaKMeansExample.java (merged with mllib/JavaKMeans.java)

AmplabJenkins commented:
Can one of the admins verify this patch?

holdenk commented:
Yay deleting code! :) Verified none of the removed examples are currently directly referenced in includes under docs/ - @yongtang still might be good for test purposes to build the docs and verify no new warnings just to be on the safe side.

srowen commented:
Jenkins test this please

SparkQA commented:
**[Test build #54926 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54926/consoleFull)** for PR 12143 at commit [`1fec5e4`](https://github.com/apache/spark/commit/1fec5e4d0adb7fd4a5c1f36a967a02dcdb1cd6e5).

SparkQA commented:
**[Test build #54926 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/54926/consoleFull)** for PR 12143 at commit [`1fec5e4`](https://github.com/apache/spark/commit/1fec5e4d0adb7fd4a5c1f36a967a02dcdb1cd6e5).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/54926/
Test PASSed.

yongtang commented:
Thanks @holdenk. Just tried with `SKIP_API=1 jekyll build` and the docs builds fine.

holdenk commented:
Awesome :) A few minor things found while reviewing:

One thing I noticed while looking at the differences between the code we are deleting and the code we are keeping is that some of the code we are deleting had a some useful javadoc on how to run the example and where it was from (e.g. `JavaTrainValidationSplitExample.java`) we might want to copy over.

Also I think the reference for what `JavaSimpleTextClassificationPipeline.java` duplicated is pointing to the wrong file since it doesn't do the cross validator stuff (although it looks like the cross validator is a super set so maybe its ok to kill anyways - but if we do kill it we should probably rename the new  to include something like "WithTextClassificationPipeline" one so people looking for a sample text pipeline 
can still find it)

jkbradley commented:
Please do not delete JavaDeveloperApiExample.java   AFAIK, this is the only code example we have for Java developers who want to build custom ML pipelines.

yongtang committed:
[SPARK-14301][Examples] Java examples code merge and clean up.

The following files have been restored:

ml/JavaDeveloperApiExample.java
ml/JavaSimpleTextClassificationPipeline.java

yongtang commented:
Thanks @jkbradley @holdenk.

Both `JavaDeveloperApiExample.java` and `JavaSimpleTextClassificationPipeline.java` have been restored.

Some of the useful comments have also been consolidated into the merged files.

srowen commented:
Jenkins retest this please.

SparkQA commented:
**[Test build #55206 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55206/consoleFull)** for PR 12143 at commit [`0b387a5`](https://github.com/apache/spark/commit/0b387a5e1c7dfc1e3cc24c38690586c307dd534c).

SparkQA commented:
**[Test build #55206 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55206/consoleFull)** for PR 12143 at commit [`0b387a5`](https://github.com/apache/spark/commit/0b387a5e1c7dfc1e3cc24c38690586c307dd534c).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:
  * `public class JavaDeveloperApiExample `
  * `    // where index i corresponds to class i (i = 0, 1).`
  * `public class JavaSimpleTextClassificationPipeline `

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/55206/
Test PASSed.

MLnick commented on examples/src/main/java/org/apache/spark/examples/ml/JavaModelSelectionViaTrainValidationSplitExample.java:
```diff
@@ -33,6 +33,16 @@
 /**
  * Java example for Model Selection via Train Validation Split.
+ *
+ * A simple example demonstrating model selection using TrainValidationSplit.
```
We can consolidate this comment line with the one above, this one is a bit redundant. Maybe something like `Java example demonstrating model selection using TrainValidationSplit"

yongtang replied:
Thanks @MLnick the comment lines have been updated.

yongtang committed:
[SPARK-14301][Examples] Java examples code merge and clean up.

Better description of comment in JavaModelSelectionViaTrainValidationSplitExample.java.

srowen commented:
Jenkins retest this please

SparkQA commented:
**[Test build #55342 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55342/consoleFull)** for PR 12143 at commit [`dff87e6`](https://github.com/apache/spark/commit/dff87e6f70fe1a7036eb4d7d2eb5469fa00b923e).

SparkQA commented:
**[Test build #55342 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55342/consoleFull)** for PR 12143 at commit [`dff87e6`](https://github.com/apache/spark/commit/dff87e6f70fe1a7036eb4d7d2eb5469fa00b923e).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/55342/
Test PASSed.

srowen commented:
Merged to master
